vizro-core/tests/unit/vizro/models/_navigation/test_navigation.py

```diff
@@ -11,51 +12,59 @@
 
 @pytest.mark.usefixtures("dashboard_prebuild")
 class TestNavigationInstantiation:
-    """Tests navigation model instantiation."""
+    """Tests navigation model instantiation ."""
 
-    def test_navigation_default(self):
-        navigation = vm.Navigation(id="navigation")
-        assert navigation.id == "navigation"
-        assert navigation.pages == ["Page 1", "Page 2"]
+    @pytest.mark.parametrize("navigation", [None, vm.Navigation()])
+    def test_navigation_default(self, page1, page2, navigation):
+        # Navigation is optional inside Dashboard and navigation.pages will always be auto-populated if not provided
+        dashboard = vm.Dashboard(pages=[page1, page2], navigation=navigation)
+        assert hasattr(dashboard.navigation, "id")
+        assert dashboard.navigation.pages == ["Page 1", "Page 2"]
 
-    def test_navigation_pages_as_list(self, pages_as_list):
+    def test_navigation_valid_pages_as_list(self, pages_as_list):
         navigation = vm.Navigation(pages=pages_as_list, id="navigation")
         assert navigation.id == "navigation"
         assert navigation.pages == pages_as_list
 
-    def test_navigation_pages_as_dict(self, pages_as_dict):
+    def test_navigation_valid_pages_as_dict(self, pages_as_dict):
         navigation = vm.Navigation(pages=pages_as_dict, id="navigation")
         assert navigation.id == "navigation"
         assert navigation.pages == pages_as_dict
 
-    def test_navigation_not_all_pages_included(self):
-        with pytest.warns(UserWarning):
-            vm.Navigation(pages=["Page 1"])
+    def test_navigation_valid_pages_not_all_included(self):
+        navigation = vm.Navigation(pages=["Page 1"], id="navigation")
+        assert navigation.id == "navigation"
+        assert navigation.pages == ["Page 1"]
+
+    def test_navigation_invalid_pages_empty_list(self):
+        with pytest.raises(ValidationError, match="Ensure this value has at least 1 item."):
+            vm.Navigation(pages=[], id="navigation")
```

Comment:
If `[None, vm.Navigation()]` works, why should this not work?

Reply:
`[None, vm.Navigation()]` are the cases we take care of, so they don't have to specify a Navigation model at all. It's the two cases covered by the auto-population of `navigation.pages` in the Dashboard validator.

However, in this case, the user explicitly configured `dashboard.navigation = vm.Navigation(pages=[])`. It's similar to our checks in other models, e.g., `Page.components `or `Dashboard.pages` where we check that the lists they provide are not empty.

Reply:
Hm I see, but I think the case is slightly different, as here we would still know what to do, whereas in the cases you described, the configuration does not make any sense. If it was easy, I'd also handle this case, but happy to skip if this is a bother